jobs/haproxy/templates/haproxy.config.erb

```diff
@@ -527,20 +563,27 @@ backend http-routed-backend-<%= prefix_hash %>
     resolvers = "resolvers default "
   end
   backend_ssl = ""
-  if data["backend_ssl"] then
-    if data["backend_ssl"].downcase == "verify" then
+
+  if (data["backend_ssl"] || "").downcase != "verify"
```

Comment:
why not fold this into one with the 
```
if data["backend_ssl"].downcase == "verify"
```
case in line 574?

i.e
```
if data["backend_ssl"]
  if data["backend_ssl"].downcase != "verify"
    (...)
  else
    (...)
```

Reply:
because there's an `elsif` on line 533 which selects out 'noverify' which means there is no existing case for _just_ `!= "verify"`

Reply:
I guess you could have 

```ruby
  if data["backend_ssl"]
    if data["backend_ssl"].downcase == "verify"
      backend_ssl = "ssl verify required ca-file /var/vcap/jobs/haproxy/config/backend-ca-certs.pem "
      if data["backend_verifyhost"]
        backend_ssl += "verifyhost #{data["backend_verifyhost"]} "
      end
      backend_ssl += alpn_config
    else 
      if data["backend_verifyhost"]
        abort "Conflicting configuration: backend_ssl must be 'verify' to use backend_verifyhost in routed_backend_servers"
      end
      if data["backend_ssl"].downcase == "noverify"
        backend_ssl = "ssl verify none "
        backend_ssl += alpn_config
      end
    end
  end
  ```
  
if you prefer.. but that's a lot of nesting.

Reply:
ok then